### ISSUE TITLE
files in uploads folder

### ISSUE THREAD

**Issue by denislepage** [jbrzusto/motusServer#408]
There are currently around 350 files left unprocessed in the shared uploads folder, dating between January and June 2018. I spot checked a few, but they are not coming up in the uploads table (maria db).

For instance, these 3 files are in uploads folder:
2669_2018-01-29T10-43-37.901_SGdata-IJmuiden.7z
2669_2018-01-29T14-51-53.551_SGdata-IJmuiden.7z
2669_2018-02-02T13-23-38.475_SGdata-IJmuiden.7z

Only this file is listed in uploads table:
2669_2018-01-29T11-28-56.446_SGdata-IJmuiden.7z

The first 2 appear to be for the same period as the one that got processed. 

- Is there a place in the mysql database where these files are listed as failed? 
- If not, what is the process that prevents these files from being reprocessed
- are some of these files still valid ones that are on hold and should be processed? How can we tell the difference?
- can we set up a cron job so older files are being deleted (say > 1 month)? This is something we are putting in place for partial uploads, and we can extend to the uploads folder as well.

**Comment by denislepage** [comment]
Thanks. The sha1 check is now implemented in our upload tool before upload starts. Files go in in the tmp subfolder while uploaded then get moved to uploads. We may be seeing less (none?) of those remnants when we turn off the sg data upload option. We're nearly there. Still addressing a few minor issues with unsupported characters in filename today. Hopefully, they're all fixed now.

Paul, do you check the sha1 again at the end of upload to confirm integrity?





-------- Original message --------

**Comment by pmorrill** [comment]
Yes - the sha1 hash is checked after file upload, and BEFORE the SG server has been signaled that it is available for processing. If the check fails, the file was corrupted during upload, and it is deleted from the /temp folder, and never transferred to the /partial folder. The SG server never hears about it at all.

The user is sent an error msg, asking them to retry.

**Comment by denislepage** [comment]
I'm trying to decide what to do with those files. Any reason why we would not want to delete them after a few days?  If not, we should set up a CRON job to do so periodically.

sgm/uploads/partial
sgm/uploads/partial/tmp

**Comment by pmorrill** [comment]
Certainly anything in the /partial/tmp directory can be deleted safely. They should all have a .part extension.

I'd say if a file upload has not been completed after 2 or 3 days, the part file can be deleted. The user just has to start over when they get back to it.

**Comment by denislepage** [comment]
Thanks. I deleted anything in sgm/uploads/partial/tmp older than 1 Aug (about 1GB of data).

There’s another 64GB of data (387 files) in sgm/uploads/partial/. Many of them appear to have been successfully imported at another time and placed in the individual receiver folders in /uploads. Some of those relate to broken accents that we have since resolved, etc. For some others, I cannot find an obvious match among the files that have been successfully processed. My question is therefore, are there any files left behind in /uploads/partial that have yet to be processed.

Here are 2 examples:

Files POC 6-2 to 8-9-18.zip and POC 5-4 to 6-2-18.zip have been uploaded on 9 Aug 2018.  From what I can tell, this has been uploaded by Susan Heath to project 120 (Western Gulf Avian Resource Assessment). There may be other data, but at least one receiver included is SG-3116BBBK0755.

2320_2018-08-09T23-06-11.039_POC 6-2 to 8-9-18.zip
2320_2018-08-09T16-27-05.546_POC 5-4 to 6-2-18.zip

From what I can tell, these 2 files are not in the SG logs. Is there a way to check that the API call was made? From my understanding of the code, if the API notification call fails, the file remains in the folder and won’t be reprocessed?

Paul, I’d like to start storing the details of the API request and response when calling process_new_upload. I have added some code in MotusAPIClient.writeSGUploads() and called the method immediately after the API call. That should simplify the process for tracking individual files against the SG logs. That might also allow us to build a failsafe for retrying notifying the API, and possibly allow us to have a list of files that have been uploaded and a link to the JobID’s.

I just pushed some new code to sandbox. Zoe, do you have any detection files that you can try uploading on sandbox?

Let me know!

**Comment by pmorrill** [comment]
Tracking the uploads to Motus is a good idea. I'd also like to see a simple tool that allows admin to view any files in the /partial folder, and delete them if their timestamp is old. I have not looked at what you did, but presumably you are storing the file name, project and time of upload, when the upload is complete (this is when files are moved to the /partial folder ad the SG system is signalled)?
